test-case/verify-ucm-bootsequence.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+##
+## Case Name: verify-ucm-bootsequence.sh
+## Preconditions:
+##    UCMv2 is installed in /usr/share/alsa/ucm2 folder
+## Description:
+##    Run alsactl init to test the initialization settings are correct or not
+## Case step:
+##    1. check the alsa-lib, alsa-utils version
+##    2. save the original amixer settings
+##    3. run alsactl init
+##    4. restore the original amixer settings
+## Expect result:
+##    alsactl init runs successfully
+##
+
+set -e
+
+# shellcheck source=case-lib/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/lib.sh
+# shellcheck source=case-lib/user.sh
+source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/user.sh
+
+# BootSequence is introduced in alsa-utils 1.2.2-16
+ALSAUTILS_VER=1.2.2-16
+# BootSequence is introduced in alsa-lib 1.2.2-44
+ALSALIB_VER=1.2.2-44
+
+func_opt_parse_option "$@"
+
+func_check_version "libasound2" "$ALSALIB_VER" || exit 1
+func_check_version "alsa-utils" "$ALSAUTILS_VER" || exit 1
+
+# alsactl init only applies to SOF card
+[[ -n "$SOFCARD" ]] || die "No SOF card"
+if cardname=$(sof-dump-status.py -s "$SOFCARD"); then
+    [[ -z "$cardname" ]] && die "Failed to get card $SOFCARD short name"
+else
+    exit 1
+fi
+
+main()
+{
+    local tmp
+    tmp=$(mktemp -d) || die "Failed to create tmp folder."
+    cd "$tmp"
+    dlogi "Checking the card $cardname UCM2 initial settings"
+    alsactl store -f tmp.conf || die "Failed to save the amixer setting"
+    dlogc "alsactl init $SOFCARD"
+    alsactl init "$SOFCARD" || die "Failed to run alsactl init"
+    alsactl restore -f tmp.conf || die "Failed to restore the amixer setting"
+}
+
+main "#@"
```

Comment:
This won't work, sorry for the earlier typo. I meant: `main "$@"`

Reply:
> This won't work, sorry for the earlier typo. I meant: `main "$@"`

Right. I will update it.